course-v2/layouts/partials/resources_header.html

```diff
@@ -1,22 +1,37 @@
-{{ $courseData := site.Data.course }}
-{{ $zenDeskUrl := "https://mitocw.zendesk.com/hc/en-us/articles/4414681093659-I-have-downloaded-an-MIT-OpenCourseWare-course-but-I-can-t-access-the-materials-How-do-I-get-started-" }}
-{{ $trimmedBaseUrl := strings.TrimSuffix "/" site.BaseURL }}
-{{ $shortId := printf "%v-%v-%v" $courseData.primary_course_number (lower $courseData.term) $courseData.year }}
-{{ $downloadCourseUrl := printf "%v/%v.zip" $trimmedBaseUrl $shortId }}
+{{- $courseData := site.Data.course -}}
+{{- $zenDeskUrl := "https://mitocw.zendesk.com/hc/en-us/articles/4414681093659-I-have-downloaded-an-MIT-OpenCourseWare-course-but-I-can-t-access-the-materials-How-do-I-get-started-" -}}
+{{- $trimmedBaseUrl := strings.TrimSuffix "/" site.BaseURL -}}
+{{- $shortId := printf "%v-%v-%v" $courseData.primary_course_number (lower $courseData.term) $courseData.year -}}
+{{- $downloadCourseUrl := printf "%v/%v.zip" $trimmedBaseUrl $shortId -}}
+{{- $headResponse := resources.GetRemote $downloadCourseUrl (dict "method" "head") -}}
+<!-- 
```

Comment:
This is a lengthy explanation but I figured it was worth leaving. The problem here in a nutshell is that if we leave off the `(dict "method" "head")` and instead make a normal `GET` request on the ZIP file, the entire thing will be downloaded during the course of the build. This could have a bad effect when building all 2700+ OCW courses and greatly increase the time it takes to rebuild the whole site. This is why I'm making a `HEAD` request here, to just get the headers on the file. Ideally I would have just liked to inspect the headers, check the type and content length, but Hugo does not support reading the response headers. It only tries to fetch the body and present that as a resource. The potential downside to doing it this way is that it's possible other URL's could return the "unexpected EOF" error if the request returns a status code of 200 with an empty body. The alternative is to do a `GET` instead and see what the fallout is from that on build times in RC.